### PR TITLE
Adds a timeout to the web credentials manager

### PR DESCRIPTION
**Pull request by orta**
This should fix the hung spinners after 5s - as the code it outside of our control.

### CODE REVIEW

**Comment by orta** [comment]
Can confirm this works, just got a re-pro locally

File: Artsy/Networking/API_Modules/ARUserManager.m [hunk 1]
```diff
@@ -602,7 +603,26 @@ - (void)tryLoginWithSharedWebCredentials:(void (^)(NSError *error))completion;
...
+        NSDictionary *info = @{NSLocalizedDescriptionKey : @"Could not connect to iCloud in a resonable time."};
+        completion([NSError errorWithDomain:@"net.artsy.artsy.authentication" code:-1 userInfo:info]);
+    });
+
     SecRequestSharedWebCredential(NULL, NULL, ^(CFArrayRef credentials, CFErrorRef error) {
+        if (hasTimedOut) {
+            return;
+        }
+
+        hasGotCredentials = YES;
```
Comment (alloy) [inline Artsy/Networking/API_Modules/ARUserManager.m]: Technically this access could be unsafe, i.e. these bools could get written while being read. Maybe us `OSAtomicIncrement32()` to change an int from 0 to 1?

Comment (ashfurrow) [reply Artsy/Networking/API_Modules/ARUserManager.m]: I would be happy if we dispatched back to the main queue – simpler, avoids race conditions, and we probably want to be there for UI updates anyway.

Comment (orta) [reply Artsy/Networking/API_Modules/ARUserManager.m]: Yeah, that works for me

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_travis">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by orta** [comment]
OK, it's now all main threaded - merge on green

**Pull request merged**